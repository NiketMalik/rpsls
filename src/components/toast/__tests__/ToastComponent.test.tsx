import React from "react";
import {
  act,
  fireEvent,
  render,
  screen,
  waitFor,
  waitForElementToBeRemoved,
} from "test-utils";

import { Toast } from "../index";

describe("Toast", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  it("renders without error", () => {
    const { container } = render(<Toast />);
    const toastWrapperEle = screen.getByTestId("toast-wrapper");
    expect(toastWrapperEle).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders basic toast", async () => {
    const { container } = render(<Toast />);
    const toastWrapperEle = screen.getByTestId("toast-wrapper");
    expect(toastWrapperEle).toBeInTheDocument();

    act(() => {
      window.showToast({
        body: "test",
      });
    });

    await waitFor(() =>
      expect(screen.getByTestId("toast")).toBeInTheDocument(),
    );
    expect(container).toMatchSnapshot();
  });

  it("renders toast with header", async () => {
    const { container } = render(<Toast />);
    const toastWrapperEle = screen.getByTestId("toast-wrapper");
    expect(toastWrapperEle).toBeInTheDocument();

    act(() => {
      window.showToast({
        body: "test",
        header: "test-header",
      });
    });

    await waitFor(() =>
      expect(screen.getByTestId("toast")).toBeInTheDocument(),
    );
    expect(container).toMatchSnapshot();
  });

  it("renders accent danger toast", async () => {
    const { container } = render(<Toast />);
    const toastEle = screen.getByTestId("toast-wrapper");
    expect(toastEle).toBeInTheDocument();

    act(() => {
      window.showToast({
        header: "test",
        body: "test",
        accent: "danger",
      });
    });

    await waitFor(() =>
      expect(screen.getByTestId("toast")).toBeInTheDocument(),
    );
    expect(container).toMatchSnapshot();
  });

  it("auto hides toast", async () => {
    const { container } = render(<Toast />);
    const toastWrapperEle = screen.getByTestId("toast-wrapper");
    expect(toastWrapperEle).toBeInTheDocument();

    act(() => {
      window.showToast({
        header: "test",
        body: "test",
      });
    });

    await waitFor(() =>
      expect(screen.getByTestId("toast")).toBeInTheDocument(),
    );

    const toastEle = screen.getByTestId("toast");
    act(() => {
      jest.runAllTimers();
    });

    expect(toastWrapperEle).not.toContainElement(toastEle);
    expect(container).toMatchSnapshot();
  });

  it("does not auto hides toast", async () => {
    const { container } = render(<Toast />);
    const toastWrapperEle = screen.getByTestId("toast-wrapper");
    expect(toastWrapperEle).toBeInTheDocument();

    act(() => {
      window.showToast({
        header: "test",
        body: "test",
        autohide: false,
      });
    });

    await waitFor(() =>
      expect(screen.getByTestId("toast")).toBeInTheDocument(),
    );

    const toastEle = screen.getByTestId("toast");
    act(() => {
      jest.runAllTimers();
    });

    expect(toastWrapperEle).toContainElement(toastEle);
    expect(container).toMatchSnapshot();
  });

  it("should hide toast on click header close click", async () => {
    render(<Toast />);
    const toastWrapperEle = screen.getByTestId("toast-wrapper");
    expect(toastWrapperEle).toBeInTheDocument();

    act(() => {
      window.showToast({
        header: "test",
        body: "test",
        autohide: false,
      });
    });

    await waitFor(() =>
      expect(screen.getByTestId("toast")).toBeInTheDocument(),
    );

    const toastEle = screen.getByTestId("toast");
    fireEvent.click(screen.getByTestId("toast-close-action"));

    expect(toastWrapperEle).toBeInTheDocument();
    expect(toastWrapperEle).not.toContainElement(toastEle);
  });
});
