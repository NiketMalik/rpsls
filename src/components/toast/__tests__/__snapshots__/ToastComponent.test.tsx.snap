// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast auto hides toast 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="polite"
    class="css-1fgetv3"
    data-testid="toast-wrapper"
  >
    <div
      class="d-flex flex-column align-items-end me-4 css-1rcgvyn"
    />
  </div>
</div>
`;

exports[`Toast does not auto hides toast 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="polite"
    class="css-1fgetv3"
    data-testid="toast-wrapper"
  >
    <div
      class="d-flex flex-column align-items-end me-4 css-1rcgvyn"
    >
      <div
        style="opacity: 0;"
      >
        <div
          aria-atomic="true"
          aria-live="assertive"
          class="fade toast bg-secondary css-x1oxl4 show"
          data-testid="toast"
          role="alert"
        >
          <div
            class="toast-header"
          >
            <strong
              class="me-auto"
            >
              test
            </strong>
            <button
              aria-label="Close"
              class="btn-close me-2 m-auto"
              data-bs-dismiss="toast"
              data-testid="toast-close-action"
              type="button"
            />
          </div>
          <div
            class="d-flex fs-5 toast-body"
          >
            <div
              class="flex-grow-1"
            >
              test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast renders accent danger toast 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="polite"
    class="css-1fgetv3"
    data-testid="toast-wrapper"
  >
    <div
      class="d-flex flex-column align-items-end me-4 css-1rcgvyn"
    >
      <div
        style="opacity: 0;"
      >
        <div
          aria-atomic="true"
          aria-live="assertive"
          class="fade toast bg-danger css-x1oxl4 show"
          data-testid="toast"
          role="alert"
        >
          <div
            class="toast-header text-white"
          >
            <strong
              class="me-auto"
            >
              test
            </strong>
            <button
              aria-label="Close"
              class="btn-close me-2 m-auto btn-close-white"
              data-bs-dismiss="toast"
              data-testid="toast-close-action"
              type="button"
            />
          </div>
          <div
            class="d-flex fs-5 text-white toast-body"
          >
            <div
              class="flex-grow-1"
            >
              test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast renders basic toast 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="polite"
    class="css-1fgetv3"
    data-testid="toast-wrapper"
  >
    <div
      class="d-flex flex-column align-items-end me-4 css-1rcgvyn"
    >
      <div
        style="opacity: 0;"
      >
        <div
          aria-atomic="true"
          aria-live="assertive"
          class="fade toast bg-secondary css-x1oxl4 show"
          data-testid="toast"
          role="alert"
        >
          <div
            class="d-flex fs-5 toast-body"
          >
            <div
              class="flex-grow-1"
            >
              test
            </div>
            <button
              aria-label="Close"
              class="btn-close me-2 m-auto"
              data-bs-dismiss="toast"
              data-testid="toast-close-action"
              type="button"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast renders toast with header 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="polite"
    class="css-1fgetv3"
    data-testid="toast-wrapper"
  >
    <div
      class="d-flex flex-column align-items-end me-4 css-1rcgvyn"
    >
      <div
        style="opacity: 0;"
      >
        <div
          aria-atomic="true"
          aria-live="assertive"
          class="fade toast bg-secondary css-x1oxl4 show"
          data-testid="toast"
          role="alert"
        >
          <div
            class="toast-header"
          >
            <strong
              class="me-auto"
            >
              test-header
            </strong>
            <button
              aria-label="Close"
              class="btn-close me-2 m-auto"
              data-bs-dismiss="toast"
              data-testid="toast-close-action"
              type="button"
            />
          </div>
          <div
            class="d-flex fs-5 toast-body"
          >
            <div
              class="flex-grow-1"
            >
              test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast renders without error 1`] = `
<div>
  <div
    aria-atomic="true"
    aria-live="polite"
    class="css-1fgetv3"
    data-testid="toast-wrapper"
  >
    <div
      class="d-flex flex-column align-items-end me-4 css-1rcgvyn"
    />
  </div>
</div>
`;
